File: POS System - Retaurant - 2022/POS System/Waiters.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace POS_System
{

    public partial class Waiters
    {
        private bool drag;
        private int mousex;
        private int mousey;
        private string EditString;
        private int Max_ID = 0;
        private DataGridViewCellStyle objCellStyle = new DataGridViewCellStyle();
        private bool Addition, Editing, RollBack;

        public Waiters()
        {
            InitializeComponent();
        }

        private void frmCompanies_Load(object sender, EventArgs e)
        {
            FormBorderStyle = FormBorderStyle.None;
            mdlGeneral.GetConnectionString();
            LoadDB();
        }

        private void BindingNavigatorQuery(string Query)
        {
            // Try
            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, Query, mdlGeneral.Connection);
            mdlGeneral.Connection.Close();
            mdlGeneral.SqlDataAdapter.Dispose();
            // Catch ex As Exception
            // MsgBox(ex.Message, MsgBoxStyle.Information, My.Application.Info.ProductName)
            // End Try
        }

        private void LoadDB()
        {

            try
            {
                CompanyDataSet.Clear();
                string CommandText = "SELECT * FROM WAITERS ORDER BY WAITER_ID";
                mdlGeneral.ExecuteDatasetQuery(mdlGeneral.SqlDataAdapter, mdlGeneral.SqlDataSet, CommandText, "WAITERS", mdlGeneral.Connection);
                mdlGeneral.SqlDataAdapter.Fill(CompanyDataSet, "WAITERS");
                CompanyBindingSource.DataSource = CompanyDataSet;
                CompanyBindingSource.DataMember = "WAITERS";
                txtID.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_ID");
                txtName.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_NAME");
                txtAddress.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_ADDRESS");
                txtPhone.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_CONTACT");
                txtCNIC.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_NIC");
                Addition = false;
                Editing = false;
                ToolStripSaveButton.Enabled = false;
                Panel1.Enabled = false;
                objCellStyle.BackColor = Color.WhiteSmoke;
                CompanyDataGridView.AlternatingRowsDefaultCellStyle = objCellStyle;
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message, MsgBoxStyle.Information, My.MyProject.Application.Info.ProductName);
                Close();
            }
            mdlGeneral.Connection.Close();

        }

        private void ToolStripCloseButton_Click(object sender, EventArgs e)
        {
            if (ToolStripCloseButton.Text == "&Close")
            {
                mdlGeneral.Connection.Close();
                CompanyBindingSource.Dispose();
                Close();
                //My.MyProject.Forms.Wholesalers.WindowState = FormWindowState.Normal;
            }
            else
            {
                ToolStripCloseButton.Text = "&Close";
                Addition = false;
                Editing = false;
                ToolStripSaveButton.Enabled = false;
                Panel1.Enabled = false;
                ToolStripEditButton.Enabled = true;
                BindingNavigatorAddNewItem.Enabled = true;
                BindingNavigatorDeleteItem.Enabled = true;
                RefreshDB();
            }
        }

        private void ToolStripSaveButton_Click(object sender, EventArgs e)
        {

            // EditString = string.empty
            if (Addition == true & Editing == false)
            {
                Max_ID = mdlGeneral.GenerateMax_ID("WAITERS", "WAITER_ID");
                if (Max_ID == 1)
                    Max_ID = Max_ID + 100;
                txtID.Text = Max_ID.ToString();
            }
            // // If changes made in selected record then execute update record query
            if (Editing == true & Addition == false)
                BindingNavigatorQuery("UPDATE WAITERS SET" + " WAITER_NAME = '" + txtName.Text.Trim().ToUpper() + "'," + " WAITER_ADDRESS = '" + txtAddress.Text.Trim().ToUpper() + "'," + " WAITER_CONTACT = '" + txtPhone.Text.Trim() + "'," + " WAITER_NIC = '" + txtCNIC.Text.Trim() + "' " + " WHERE WAITER_ID = " + Conversion.Val(EditString));





            // // if entered new values in text boxes then execute add record query
            if (Addition == true & Editing == false)
                BindingNavigatorQuery("INSERT INTO WAITERS VALUES(" + Conversion.Val(txtID.Text.Trim()) + ",'" + txtName.Text.Trim().ToUpper() + "','" + txtAddress.Text.Trim().ToUpper() + "','" + txtPhone.Text.Trim() + "','" + txtCNIC.Text.Trim() + "')");




            RefreshDB();
            ToolStripSaveButton.Enabled = false;
            Panel1.Enabled = false;
            Editing = false;
            Addition = false;
            ToolStripEditButton.Enabled = true;
            ToolStripCloseButton.Text = "&Close";

        }

        private void BindingNavigatorAddNewItem_Click(object sender, EventArgs e)
        {

            Panel1.Enabled = true;
            txtName.Focus();
            Addition = true;
            RollBack = true;
            Editing = false;
            EditString = string.Empty;
            ToolStripSaveButton.Enabled = true;
            ToolStripCloseButton.Text = "&Cancel";

        }

        private void ToolStripEditButton_Click(object sender, EventArgs e)
        {
            Editing = true;
            Addition = false;
            Panel1.Enabled = true;
            ToolStripSaveButton.Enabled = true;
            ToolStripCloseButton.Text = "&Cancel";
            EditString = txtID.Text.Trim();
            txtName.Focus();

        }

        private void BindingNavigatorDeleteItem_Click(object sender, EventArgs e)
        {

            if (Interaction.MsgBox("Are you sure to delete ?" + Strings.Space(2) + "(" + Strings.Trim(txtName.Text.ToUpper()) + ")", (MsgBoxStyle)((int)MsgBoxStyle.Information + (int)MsgBoxStyle.OkCancel), "Delete Record") == MsgBoxResult.Ok)
            {
                string CommandText = "DELETE FROM WAITERS WHERE WAITER_ID = " + Conversion.Val(txtID.Text.Trim());
                mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, CommandText, mdlGeneral.Connection);
                CompanyDataGridView.Update();
            }
            else
            {
                // skip
                RefreshDB();
            }

        }
        private void RefreshDB()
        {

            try
            {
                CompanyDataSet.Clear();
                mdlGeneral.SqlDataAdapter.Dispose();
                string QueryText = "SELECT * FROM WAITERS ORDER BY WAITER_ID";
                mdlGeneral.ExecuteDatasetQuery(mdlGeneral.SqlDataAdapter, mdlGeneral.SqlDataSet, QueryText, "WAITERS", mdlGeneral.Connection);
                mdlGeneral.SqlDataAdapter.Fill(CompanyDataSet, "WAITERS");
                CompanyBindingSource.DataSource = CompanyDataSet;
                CompanyBindingSource.DataMember = "WAITERS";
                CompanyDataGridView.Update();
                CompanyDataGridView.RefreshEdit();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message, MsgBoxStyle.Information, My.MyProject.Application.Info.ProductName);
                Close();
            }
            mdlGeneral.Connection.Close();

        }

        private void ToolStripCloseButton_MouseDown(object sender, MouseEventArgs e)
        {
            drag = true;
            mousex = Cursor.Position.X - Left;
            mousey = Cursor.Position.Y - Top;
        }

        private void ToolStripCloseButton_MouseMove(object sender, MouseEventArgs e)
        {
            if (drag)
            {
                Top = Cursor.Position.Y - mousey;
                Left = Cursor.Position.X - mousex;
            }
        }

        private void ToolStripCloseButton_MouseUp(object sender, MouseEventArgs e)
        {
            drag = false;
        }
    }
}

File: POS System - Retaurant/POS System/Waiters.vb
Imports System.Data.Sql
Imports System.Data.SqlClient

Public Class Waiters
    Dim drag As Boolean
    Dim mousex As Integer
    Dim mousey As Integer
    Dim EditString As String
    Dim Max_ID As Integer = 0
    Dim objCellStyle As New DataGridViewCellStyle()
    Dim Addition, Editing, RollBack As Boolean

    Private Sub frmCompanies_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.FormBorderStyle = Windows.Forms.FormBorderStyle.None
        Call GetConnectionString()
        Call LoadDB()
    End Sub

    Private Sub BindingNavigatorQuery(ByVal Query As String)
        'Try
        ExecuteAdapterQuery(SqlDataAdapter, Query, Connection)
        Connection.Close()
        SqlDataAdapter.Dispose()
        'Catch ex As Exception
        '    MsgBox(ex.Message, MsgBoxStyle.Information, My.Application.Info.ProductName)
        'End Try
    End Sub

    Private Sub LoadDB()

        Try
            Me.CompanyDataSet.Clear()
            Dim CommandText As String = "SELECT * FROM WAITERS ORDER BY WAITER_ID"
            ExecuteDatasetQuery(SqlDataAdapter, SqlDataSet, CommandText, "WAITERS", Connection)
            SqlDataAdapter.Fill(Me.CompanyDataSet, "WAITERS")
            Me.CompanyBindingSource.DataSource = Me.CompanyDataSet
            Me.CompanyBindingSource.DataMember = "WAITERS"
            Me.txtID.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_ID")
            Me.txtName.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_NAME")
            Me.txtAddress.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_ADDRESS")
            Me.txtPhone.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_CONTACT")
            Me.txtCNIC.DataBindings.Add("TEXT", CompanyBindingSource, "WAITER_NIC")
            Addition = False
            Editing = False
            ToolStripSaveButton.Enabled = False
            Panel1.Enabled = False
            objCellStyle.BackColor = Color.WhiteSmoke
            CompanyDataGridView.AlternatingRowsDefaultCellStyle = objCellStyle
        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Information, My.Application.Info.ProductName)
            Me.Close()
        End Try
        Connection.Close()

    End Sub

    Private Sub ToolStripCloseButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripCloseButton.Click
        If ToolStripCloseButton.Text = "&Close" Then
            Connection.Close()
            Me.CompanyBindingSource.Dispose()
            Me.Close()
            Wholesalers.WindowState = FormWindowState.Normal
        Else
            ToolStripCloseButton.Text = "&Close"
            Addition = False
            Editing = False
            ToolStripSaveButton.Enabled = False
            Panel1.Enabled = False
            ToolStripEditButton.Enabled = True
            BindingNavigatorAddNewItem.Enabled = True
            BindingNavigatorDeleteItem.Enabled = True
            Call RefreshDB()
        End If
    End Sub

    Private Sub ToolStripSaveButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripSaveButton.Click

        'EditString = string.empty
        If Addition = True And Editing = False Then
            Max_ID = GenerateMax_ID("WAITERS", "WAITER_ID")
            If Max_ID = 1 Then Max_ID = Max_ID + 100
            txtID.Text = Max_ID
        End If
        '// If changes made in selected record then execute update record query
        If Editing = True And Addition = False Then Call BindingNavigatorQuery("UPDATE WAITERS SET" & _
                                                           " WAITER_NAME = '" & txtName.Text.Trim.ToUpper & "'," & _
                                                           " WAITER_ADDRESS = '" & txtAddress.Text.Trim.ToUpper & "'," & _
                                                           " WAITER_CONTACT = '" & txtPhone.Text.Trim & "'," & _
                                                           " WAITER_NIC = '" & txtCNIC.Text.Trim & "' " & _
                                                           " WHERE WAITER_ID = " & Val(EditString))

        '// if entered new values in text boxes then execute add record query
        If Addition = True And Editing = False Then Call BindingNavigatorQuery("INSERT INTO WAITERS VALUES(" & Val(txtID.Text.Trim) & ",'" & _
                                                           txtName.Text.Trim.ToUpper & "','" & _
                                                           txtAddress.Text.Trim.ToUpper & "','" & _
                                                           txtPhone.Text.Trim & "','" & _
                                                           txtCNIC.Text.Trim & "')")

        Call RefreshDB()
        ToolStripSaveButton.Enabled = False
        Panel1.Enabled = False
        Editing = False
        Addition = False
        ToolStripEditButton.Enabled = True
        Me.ToolStripCloseButton.Text = "&Close"

    End Sub

    Private Sub BindingNavigatorAddNewItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorAddNewItem.Click

        Panel1.Enabled = True
        txtName.Focus()
        Addition = True
        RollBack = True
        Editing = False
        EditString = String.Empty
        ToolStripSaveButton.Enabled = True
        ToolStripCloseButton.Text = "&Cancel"

    End Sub

    Private Sub ToolStripEditButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripEditButton.Click
        Editing = True
        Addition = False
        Panel1.Enabled = True
        ToolStripSaveButton.Enabled = True
        ToolStripCloseButton.Text = "&Cancel"
        EditString = txtID.Text.Trim
        txtName.Focus()

    End Sub

    Private Sub BindingNavigatorDeleteItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorDeleteItem.Click

        If MsgBox("Are you sure to delete ?" & Space(2) & "(" & Trim(txtName.Text.ToUpper) & ")", MsgBoxStyle.Information + MsgBoxStyle.OkCancel, "Delete Record") = MsgBoxResult.Ok Then
            Dim CommandText As String = "DELETE FROM WAITERS WHERE WAITER_ID = " & Val(txtID.Text.Trim)
            ExecuteAdapterQuery(SqlDataAdapter, CommandText, Connection)
            Me.CompanyDataGridView.Update()
        Else
            'skip
            Call RefreshDB()
        End If

    End Sub
    Private Sub RefreshDB()

        Try
            Me.CompanyDataSet.Clear()
            SqlDataAdapter.Dispose()
            Dim QueryText As String = "SELECT * FROM WAITERS ORDER BY WAITER_ID"
            ExecuteDatasetQuery(SqlDataAdapter, SqlDataSet, QueryText, "WAITERS", Connection)
            SqlDataAdapter.Fill(Me.CompanyDataSet, "WAITERS")
            Me.CompanyBindingSource.DataSource = Me.CompanyDataSet
            Me.CompanyBindingSource.DataMember = "WAITERS"
            Me.CompanyDataGridView.Update()
            Me.CompanyDataGridView.RefreshEdit()
        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Information, My.Application.Info.ProductName)
            Me.Close()
        End Try
        Connection.Close()

    End Sub

    Private Sub ToolStripCloseButton_MouseDown(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles ToolStripCloseButton.MouseDown
        drag = True
        mousex = Windows.Forms.Cursor.Position.X - Me.Left
        mousey = Windows.Forms.Cursor.Position.Y - Me.Top
    End Sub

    Private Sub ToolStripCloseButton_MouseMove(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles ToolStripCloseButton.MouseMove
        If drag Then
            Me.Top = Windows.Forms.Cursor.Position.Y - mousey
            Me.Left = Windows.Forms.Cursor.Position.X - mousex
        End If
    End Sub

    Private Sub ToolStripCloseButton_MouseUp(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles ToolStripCloseButton.MouseUp
        drag = False
    End Sub
End Class